vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+tr <- derive_vars_merged(
+  tr,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = vars(STUDYID, USUBJID)
+)
+```
+
+## Merge TR with TU and derive new variables {#merge}
+
+Depending on your company specifications you may want to merge `TU` with `TR`.
+As an example, here we are keeping the tumor location, and deriving a new
+variable for the tumor group site. You may want to process other variables like
+`ADT`.
+
+```{r}
+tr <- derive_vars_merged(
+  tr,
+  dataset_add = rename(tu, TRLNKID = TULNKID),
+  new_vars = vars(TULOC),
+  by_vars = vars(STUDYID, USUBJID, TRLNKID)
+) %>% mutate(
+  TULOCGR1 = if_else(
+    TULOC == "LYMPH NODE",
+    "NODAL",
+    "NON-NODAL"
+  )
+)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(tr, display_vars = vars(USUBJID, TRLNKID, TULOC, TULOCGR1))
+```  
+
+
+In addition, you could create any additional new variables that are required
+downstream when deriving the required parameters. For example, here we include
+lesion ID expected and lesion ID assessed. (any other examples?)
+
+```{r}
+tr <- mutate(
+  tr,
+  LSEXP = TRLNKID,
+  LSASS = if_else(!is.na(TRSTRESN), TRLNKID, NA_character_)
+)
+```
+
+## Pre-processing of input records {#input}
+
+The next step involves company-specific pre-processing of records for the
+required input to downstream parameter derivations.  See examples below.
+
+### Partial Date Imputation and Deriving `ADT`, `ADTF`, `AVISIT` etc
+
+If your data collection allows for partial dates, you could apply a
+company-specific imputation rule at this stage when deriving `ADT`. For this
+example, here we impute missing day to earliest possible date.
+
+```{r}
+tr <- derive_vars_dt(
+  tr,
+  dtc = TRDTC,
+  new_vars_prefix = "A",
+  highest_imputation = "D",
+  date_imputation = "first"
+) %>%
+  derive_vars_dy(
+    reference_date = RANDDT,
+    source_vars = vars(ADT)
+  )
+```
+
+
+```{r, echo=FALSE}
+dataset_vignette(tr, display_vars = vars(USUBJID, RANDDT, TRLNKID, TRDTC, ADT, ADTF, ADY))
+```  
+
+### Re-labelling unscheduled visits
+
+If you have data collected at unscheduled visits, which are not identified as
+unique in your SDTM or you have company specific rules.  Below is an example of
+how it could be re-labelled. <add more information once I see the template code>
+
+```{r}
+tr <- mutate(
+  tr,
+  AVISIT = VISIT,
+  AVISITN = VISITNUM
+)
+```
+
+## Derive Parameters TLDn and TLPn {#parameters}
+
+As mentioned earlier, depending on your company specifications you may either
+create parameters for TLDn and TLPn in ADTR or similar dataset.
+
+Set the `AVALx/ANL01FL` variables.
+```{r}
+adtr <- tr %>%
+  mutate(
+    tmp_lesion_nr = str_sub(TRLNKID, 3),
+    PARAMCD = paste0("DIAM", tmp_lesion_nr),
+    PARAM = paste("Target Lesion", tmp_lesion_nr, "Analysis Diameter by Investigator"),
+    PARCAT1 = "Target Lesion(s)",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "Recist 1.1",
+    AVAL = TRSTRESN,
+    ANL01FL = if_else(!is.na(AVAL), "Y", NA_character_)
```

Comment:
**[From Catherine Jin via Stephen Gormley]**

This ANL01FL derivation does not consider PD or any new anti-cancer therapy, also does not exclude any pre-baseline records.  Does this require some notification that companies should update the code for this specific ANL01FL derivation (given this flag is the base for several other parameters). Perhaps add something in the header before the chunk, e.g. _depending on your company specifications......_

Reply:
Hi Catherine, could you explain a bit more about excluding pre-baseline records? 
We discussed to keep ANL01FL as basic then building up on the derivation, ANL03/4FL considers PD.  We are talking about new -anti cancer therapy so perhaps we could tie this in as another example later once we have discussed issue #17 ?  I can explain a bit more in the text about how we approached it, but I saw these as more examples and for each company to update the code based on specific needs.